tomcat/src/main/java/com/linecorp/armeria/server/tomcat/UnmanagedTomcatService.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.tomcat;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.apache.catalina.connector.Connector;
+import org.apache.catalina.startup.Tomcat;
+
+class UnmanagedTomcatService extends TomcatService {
+    @Nullable
+    private final Tomcat tomcat;
+    @Nullable
+    private final String hostName;
+    @Nullable
+    private final Connector connector;
+
+    UnmanagedTomcatService(Tomcat tomcat) {
+        this.hostName = null;
+        this.connector = null;
+        this.tomcat = tomcat;
+    }
+
+    UnmanagedTomcatService(@Nullable String hostName, Connector connector) {
+        this.hostName = hostName;
+        this.connector = connector;
+        this.tomcat = null;
+    }
+
+    @Override
+    public Optional<Connector> connector() {
+        if (connector != null) {
+            return Optional.of(connector);
```

Comment:
How about storing the `Optional`s as fields instead of Nullable since we need to create `Optional` either way (I guess there's no conceivable usage where we won't create `Optional`)

Reply:
Thats a good idea